zebra-rpc/src/methods/tests/snapshot.rs

```diff
@@ -0,0 +1,242 @@
+
+    // `getblock`
+    let get_block = rpc
+        .get_block("1".to_string(), 0u8)
+        .await
+        .expect("We should have a GetBlock struct");
+    snapshot_rpc_getblock(get_block, network);
+
+    // `getbestblockhash`
+    let get_best_block_hash = rpc
+        .get_best_block_hash()
+        .expect("We should have a GetBestBlockHash struct");
+    snapshot_rpc_getbestblockhash(get_best_block_hash, network);
+
+    // `getrawmempool`
+    //
+    // - a request to get all mempool transactions will be made by `getrawmempool` behind the scenes.
+    // - as we have the mempool mocked we need to expect a request and wait for a response,
+    // which will be an empty mempool in this case.
+    let mempool_req = mempool
+        .expect_request_that(|_request| true)
+        .map(|responder| {
+            responder.respond(mempool::Response::TransactionIds(
+                std::collections::HashSet::new(),
+            ));
+        });
+
+    // make the api call
+    let get_raw_mempool = rpc.get_raw_mempool();
+    let (response, _) = futures::join!(get_raw_mempool, mempool_req);
+    let get_raw_mempool = response.expect("We should have a GetRawTransaction struct");
+
+    snapshot_rpc_getrawmempool(get_raw_mempool, network);
+
+    // `getrawtransaction`
+    //
+    // - similar to `getrawmempool` described above, a mempool request will be made to get the requested
+    // transaction from the mempoo, response will be empty as we have this transaction in state
+    let mempool_req = mempool
+        .expect_request_that(|_request| true)
+        .map(|responder| {
+            responder.respond(mempool::Response::Transactions(vec![]));
+        });
+
+    // make the api call
+    let get_raw_transaction =
+        rpc.get_raw_transaction(first_block_first_transaction.hash().encode_hex(), 0u8);
+    let (response, _) = futures::join!(get_raw_transaction, mempool_req);
+    let get_raw_transaction = response.expect("We should have a GetRawTransaction struct");
+
+    snapshot_rpc_getrawtransaction(get_raw_transaction, network);
+
+    // `getaddresstxids`
+    let get_address_tx_ids = rpc
+        .get_address_tx_ids(GetAddressTxIdsRequest {
+            addresses: addresses.clone(),
+            start: 1,
+            end: 10,
+        })
+        .await
+        .expect("We should have an vector of strings");
+    snapshot_rpc_getaddresstxids(get_address_tx_ids, network);
+
+    // `getaddressutxos`
+    let get_address_utxos = rpc
+        .get_address_utxos(AddressStrings { addresses })
+        .await
+        .expect("We should have an vector of strings");
+    snapshot_rpc_getaddressutxos(get_address_utxos, network);
+}
+
+/// Snapshot `getinfo` response, using `cargo insta` and RON serialization.
+fn snapshot_rpc_getinfo(info: GetInfo, network: Network) {
+    let settings = insta::Settings::clone_current();
+    let snap_name = format!("{}_get_info", network_string(network));
```

Comment:
Consistency: Can we use the same file name format as the state snapshots?

This is how we add a suffix with the network and height (which is always 10 in these tests):
https://github.com/ZcashFoundation/zebra/blob/7fccd6b9ee32cc8fce39466a27c8781484323966/zebra-state/src/service/finalized_state/zebra_db/block/tests/snapshot.rs#L199

And here is what the filenames look like:
https://github.com/ZcashFoundation/zebra/tree/main/zebra-state/src/service/finalized_state/zebra_db/block/tests/snapshots